presto-spi/src/main/java/com/facebook/presto/spi/block/BlockUtil.java

```diff
@@ -216,4 +228,88 @@ public static boolean internalPositionInRange(int internalPosition, int offset,
         assert withinRange : format("internalPosition %s is not within range [%s, %s)", internalPosition, offset, positionCount + offset);
         return withinRange;
     }
+
+    static boolean[] appendNullToIsNullArray(boolean[] isNull, int offsetBase, int positionCount)
+    {
+        checkArrayRange(isNull, offsetBase, positionCount);
+
+        int desiredLength = offsetBase + positionCount + 1;
+        boolean[] newIsNull = new boolean[desiredLength];
+        if (isNull != null) {
+            System.arraycopy(isNull, 0, newIsNull, 0, desiredLength - 1);
+        }
+        newIsNull[desiredLength - 1] = true;
+        return newIsNull;
+    }
+
+    static int[] appendNullToOffsetsArray(int[] offsets, int offsetBase, int positionCount)
+    {
+        checkArrayRange(offsets, offsetBase, positionCount + 1);
+
+        int desiredLength = offsetBase + positionCount + 2;
+        int[] newOffsets = Arrays.copyOf(offsets, desiredLength);
+        newOffsets[desiredLength - 1] = newOffsets[desiredLength - 2];
+        return newOffsets;
+    }
+
+    public static byte[] ensureCapacity(byte[] buffer, int capacity)
```

Comment:
Add a comment that explains that this method will preserve the original data.

Reply:
Added the following comment:
```
/**
     * Returns a short array of size capacity if the input buffer is null or smaller than the capacity.
     * If the input buffer is not null, the original values in buffer will be preserved.
     */
```